inc/mod/pages.php

```diff
@@ -2259,8 +2259,6 @@ function mod_deletebyip($boardName, $post, $global = false) {
 
         rebuildThemes('post-delete', $board['uri']);
 
-        buildIndex();
```

Comment:
There was this really suspicious line that I suspect is the one that was causing delays when deleting by IP. If you check a few lines below this, it re-builds index(). I see no reason why the index needs to be rebuilt per post deleted.

Reply:
It might even be better to collect all board accessed and do a final rebuild at the end. 

```php
$boards_to_rebuild = array();
while ($post = $query->fetch(PDO::FETCH_ASSOC)) {
    $boards_to_rebuild[$post['board']] = true; 
    openBoard($post['board']);
// ...
foreach(array_keys($boards_to_rebuild) as $board) {
    openBoard($board);
    buildIndex();
}

if ($global) {
//...
```

Reply:
Even looking at rebuild of threads - you will open the board for each and every thread to rebuild even if you are deleting threads in same board.